src/Player.js

```diff
@@ -0,0 +1,393 @@
+      }, 250);
+
+      return clip.play() || Promise.resolve();
+    } catch (err) {
+      this.onError(err);
+      return Promise.reject(err.toString());
+    }
+  }
+
+  _getClip(
+    url,
+    fileSize,
+    initialPosition = 0,
+    lastAllowedPosition = 1,
+    audioMetadata = {}
+  ) {
+    if (typeof url !== 'string') {
+      throw new ProtonPlayerError('Invalid URL');
+    }
+
+    if (typeof fileSize !== 'number') {
+      throw new ProtonPlayerError('Invalid file size');
+    }
+
+    if (this.clips[url]) {
+      return this.clips[url];
+    }
+
+    const clip = new Clip({
+      url,
+      fileSize,
+      initialPosition,
+      lastAllowedPosition,
+      audioMetadata,
+      silenceChunks: this.silenceClipState.chunks,
+      volume: this.volume,
+      osName: this.osName,
+      browserName: this.browserName,
+      useMediaSource: canUseMediaSource(),
+    });
+
+    clip.on('loaderror', (err) => {
+      error('Clip failed to load', err);
+    });
+
+    clip.on('playbackerror', (err) => {
+      error('Something went wrong during playback', err);
+    });
+
+    this.clips[url] = clip;
+    return clip;
+  }
+
+  stopAll() {
+    this.currentlyPlaying = null;
+    this._clearIntervals();
+    Object.keys(this.clips).forEach((k) => {
+      this.clips[k].offAll('loadprogress');
+      this.clips[k].stop();
+    });
+  }
+
+  resume() {
+    if (this.currentlyPlaying && this.currentlyPlaying.clip) {
+      this.currentlyPlaying.clip.resume();
+    }
+  }
+
+  pause() {
+    if (this.currentlyPlaying && this.currentlyPlaying.clip) {
+      this.currentlyPlaying.clip.pause();
+    }
+  }
+
+  setVolume(volume = 1) {
+    this.volume = volume;
+    Object.keys(this.clips).forEach((k) => {
+      this.clips[k].volume = this.volume;
+    });
+  }
+
+  setPlaybackPosition(percent, newLastAllowedPosition = null) {
+    if (!this.currentlyPlaying || percent > 1) {
+      return Promise.resolve();
+    }
+
+    this.currentlyPlaying.lastReportedProgress = percent;
+
+    const {
+      url,
+      fileSize,
+      onBufferChange,
+      onBufferProgress,
+      onPlaybackProgress,
+      onPlaybackEnded,
+      lastAllowedPosition,
+    } = this.currentlyPlaying;
+
+    newLastAllowedPosition = newLastAllowedPosition || lastAllowedPosition;
+
+    const clip = this.clips[url];
+
+    if (clip) {
+      return (
+        clip.setCurrentPosition(percent, newLastAllowedPosition) || Promise.resolve()
+      );
+    }
+
+    const audioMetadata = clip && clip.audioMetadata;
+
+    this.dispose(url);
+
+    return this.playTrack({
+      url,
+      fileSize,
+      onBufferChange,
+      onBufferProgress,
+      onPlaybackProgress,
+      onPlaybackEnded,
+      audioMetadata,
+      initialPosition: percent,
+      lastAllowedPosition: newLastAllowedPosition,
+      fromSetPlaybackPosition: true,
+    });
+  }
+
+  _dispose(url) {
+    if (this.currentlyPlaying && this.currentlyPlaying.url === url) {
```

Comment:
This could be simplified to:
```suggestion
    if (this.currentlyPlaying?.url === url) {
```